File: 4 Семестр/ООП/HelpfulFiles/CourseWorkHelpers/Процедурка чья-то/Point.cs
namespace Procedure
{
    public class Point
    {
        public double X { get; private set; }
        public double Y { get; private set; }
        public Point()
        {
            X = 0;
            Y = 0;
        }
        public Point(double x, double y)
        {
            X = x;
            Y = y;
        }
        public Point(Point p)
        {
            X = p.X;
            Y = p.Y;
        }
    }
}

File: 4 Семестр/ООП/Подготовка к зачёту/Зачёт/Point.cpp
#include "Point.h"
#include <iostream>

using namespace std;

Point::Point()
{
	x = 0;
	y = 0;
}
Point::Point(int x, int y)
{
	this->x = x;
	this->y = y;
}
Point::Point(const Point& p)
{
	x = p.x;
	y = p.y;
}

Point& Point::operator=(const Point& p)
{
	x = p.x;
	y = p.y;
	return *this;
}
const Point Point::operator+(const Point& p) const
{
	return Point(x + p.x, y + p.y);
}
Point Point::Move(int x, int y)
{
	return Point(this->x + x, this->y + y);
}
ostream& operator<<(ostream& os, const Point& p)
{
	return os << "x = " << p.x << "y = " << p.y << ";";
}